builtin/providers/aws/resource_aws_dc_intra_virtual_interface.go

```diff
@@ -0,0 +1,284 @@
+		}
+
+		log.Printf("[DEBUG] Creating DirectConnect private virtual interface")
+		resp, err = conn.AllocatePrivateVirtualInterface(createOpts)
+		if err != nil {
+			return fmt.Errorf("Error creating DirectConnect Virtual Interface: %s", err)
+		}
+
+	}
+
+	// Store the ID
+	VirtualInterface := resp
+	d.SetId(*VirtualInterface.VirtualInterfaceId)
+	log.Printf("[INFO] VirtualInterface ID: %s", *VirtualInterface.VirtualInterfaceId)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"pending"},
+		Target:     []string{"available", "confirming", "verifying", "pending"},
+		Refresh:    DirectConnectIntraVirtualInterfaceRefreshFunc(conn, *VirtualInterface.VirtualInterfaceId),
+		Timeout:    10 * time.Minute,
+		Delay:      10 * time.Second,
+		MinTimeout: 10 * time.Second,
+	}
+
+	_, stateErr := stateConf.WaitForState()
+	if stateErr != nil {
+		return fmt.Errorf(
+			"Error waiting for DirectConnect PrivateVirtualInterface (%s) to become ready: %s",
+			*VirtualInterface.VirtualInterfaceId, err)
+	}
+
+	// Read off the API to populate our RO fields.
+	return resourceAwsDirectConnectIntraVirtualInterfaceRead(d, meta)
+}
+
+func DirectConnectIntraVirtualInterfaceRefreshFunc(conn *directconnect.DirectConnect, virtualinterfaceId string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+
+		resp, err := conn.DescribeVirtualInterfaces(&directconnect.DescribeVirtualInterfacesInput{
+			VirtualInterfaceId: aws.String(virtualinterfaceId),
+		})
+
+		if err != nil {
+
+			log.Printf("Error on DirectConnectPrivateVirtualInterfaceRefresh: %s", err)
+			return nil, "", err
+
+		}
+
+		if resp == nil || len(resp.VirtualInterfaces) == 0 {
+			return nil, "", nil
+		}
+
+		virtualInterface := resp.VirtualInterfaces[0]
+		return virtualInterface, *virtualInterface.VirtualInterfaceState, nil
+	}
+}
+
+func resourceAwsDirectConnectIntraVirtualInterfaceRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dcconn
+
+	resp, err := conn.DescribeVirtualInterfaces(&directconnect.DescribeVirtualInterfacesInput{
+		VirtualInterfaceId: aws.String(d.Id()),
+	})
+
+	if err != nil {
+
+		log.Printf("[ERROR] Error finding DirectConnect PrivateVirtualInterface: %s", err)
+		return err
+
+	}
+
+	if len(resp.VirtualInterfaces) != 1 {
+		return fmt.Errorf("[ERROR] Error finding DirectConnect PrivateVirtualInterface: %s", d.Id())
+	}
+
+	virtualInterface := resp.VirtualInterfaces[0]
+
+	// Set attributes under the user's control.
+	d.Set("connection_id", *virtualInterface.ConnectionId)
+	d.Set("asn", *virtualInterface.Asn)
+	d.Set("virtual_interface_name", *virtualInterface.VirtualInterfaceName)
+	d.Set("vlan", *virtualInterface.Vlan)
+	d.Set("amazon_address", *virtualInterface.AmazonAddress)
+	d.Set("customer_address", *virtualInterface.CustomerAddress)
+	// d.Set("auth_key", *virtualInterface.AuthKey)
+
+	// Set read only attributes.
+	d.SetId(*virtualInterface.VirtualInterfaceId)
```

Comment:
Is there a chance that the Id can change between the Create and Read? We have already set it in Create